common/lib/xmodule/xmodule/partitions/partitions_service.py

```diff
@@ -3,41 +3,114 @@
 user partitions.  It uses the user_service key/value store provided by the LMS runtime to
 persist the assignments.
 """
-from abc import ABCMeta, abstractproperty
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+import logging
+
+from xmodule.partitions.partitions import UserPartition, UserPartitionError
+from xmodule.modulestore.django import modulestore
+
+
+log = logging.getLogger(__name__)
+
+
+# UserPartition IDs must be unique. The Cohort and Random UserPartitions (when they are
+# created via Studio) choose an unused ID in the range of 100 to MAX_INT. Therefore the
+# dynamic UserPartitionIDs must be under 100, and they have to be hard-coded to ensure
+# they are always the same whenever the dynamic partition is added (since the UserPartition
+# ID is stored in the xblock group_access dict).
+ENROLLMENT_TRACK_PARTITION_ID = 50
+
+
+# settings will not be available when running nosetests.
+FEATURES = getattr(settings, 'FEATURES', {})
```

Comment:
This is the same thing that was recently introduced into capa, so hopefully it is an acceptable pattern.

https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/capa_base.py#L45